models/productDao.js

```diff
@@ -16,11 +16,125 @@ const getProducts = async () => {
     JOIN location AS l ON p.location_id = l.id
     JOIN location_groups AS lg ON l.location_group_id = lg.id
     JOIN events AS e ON e.product_id = p.id
-    GROUP BY e.title;
+    GROUP BY e.title
   `);
   return getEventProduct;
 }
 
+const mainCategoryFiltering = async (mainCategoryName, sort) => {
+  const orderByobj = {
+    desc : '(p.price * (100 - p.discount_rate)) DESC',
+    asc : '(p.price * (100 - p.discount_rate)) ASC'
+  }
+  const descAndAsc = sort ? orderByobj[sort] : 'p.id';
+
+  const filtering = appDataSource.query(`
```

Comment:
await 키워드를 통해 비동기 처리를 해주세요~!
```suggestion
  const filtering = await appDataSource.query(`
```